packages/nouns-contracts/contracts/NounsAuctionHousePreV2Migration.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title An interim contract for storage migration between V1 and V2
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+import { PausableUpgradeable } from '@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol';
+import { ReentrancyGuardUpgradeable } from '@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol';
+import { OwnableUpgradeable } from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import { INounsAuctionHouse } from './interfaces/INounsAuctionHouse.sol';
+
+contract NounsAuctionHousePreV2Migration is PausableUpgradeable, ReentrancyGuardUpgradeable, OwnableUpgradeable {
+    // See NounsAuctionHouse for docs on these state vars
+    address public nouns;
+    address public weth;
+    uint256 public timeBuffer;
+    uint256 public reservePrice;
+    uint8 public minBidIncrementPercentage;
+    uint256 public duration;
+    INounsAuctionHouse.Auction public auction;
+
+    function migrate() public onlyOwner {
+        INounsAuctionHouse.Auction memory _auction = auction;
+
+        auction = INounsAuctionHouse.Auction(0, 0, 0, 0, payable(address(0)), false);
+
+        INounsAuctionHouse.AuctionV2 storage auctionV2;
+        assembly {
+            auctionV2.slot := auction.slot
```

Comment:
@eladmallel any particular reason to reuse the same slot?
alternatively we could keep the old `auction` slot and mark it as deprecated.
so I see the reason could be to keep the V2 contract cleaner. is that the reasoning or is there something else?

Reply:
definitely the reason is to keep V2 cleaner